docs/okeps/okep-5094-layer2-transit-router.md

```diff
@@ -0,0 +1,910 @@
+# OKEP-5094: Primary UDN Layer2 topology improvements
+
+* Issue: [#5094](https://github.com/ovn-kubernetes/ovn-kubernetes/issues/5094)
+
+## Problem Statement
+
+The primary UDN layer2 topology present some problems related to VM's live migration that are being addressed by
+ovn-kubernetes sending GARPs or unsolicited router advertisement and blocking some OVN router advertisement, although this fixes the issue, is not the most robust way to address the problem and adds complexity to ovn-kubernetes live migration mechanism.
+
+There are other layer2 egressip [limitations](https://issues.redhat.com/browse/OCPBUGS-48301) introduced by this topology, EIP for L2 UDN depends on a single external GW IP therefore we dont respect multiple ext gws and forward all traffic to one GW.
```

Comment:
for context, its because the routing decision is made at the gw router (instead of the cluster router) and we wanted to ensure load balancing works for local pod if there are multiple IPs assigned for one EIP obj, therefore we needed a next hop and we choose the ext gw for now. This has limitations. We dont want to depend on external IP. We dont want to depend on having an ext gw (isolated clusters). We want to respect multiple GWs as you said. Tim has context on this.

Reply:
e2es? not that im aware of

Reply:
> multiple IPs assigned for 1 obj -> load balancing here is multiple next hops right? cause from the same object only 1 IP is assigned per node, so I don't get why the next hip is the ext GW? Ah do you mean instead of the joinIP that we normally use for local hop, in L2 we actually use the external switch port rtoe's IP as the nexthop? :o
ditto as above, We need to expand on the problem in depth for better history tracking and avoid linking redhat bug links in upstream project.

@tssurya We have LRPs attached to redirect local pods packets to egress nodes but if the local node is also the egress node, and theres multiple EIP IPs for LB'ing then we choose the next hop to egress locally as default external gateway routers IP.. we want to remove that dep.

Reply:
@kyrtapz @qinqon @martinkennelly will help with testing egressip fix and providing a snippet